src/scripts/modules/media/tabs/contents/toc/section.coffee

```diff
@@ -24,37 +26,31 @@ define (require) ->
       @regions =
         container: @itemViewContainer
       @sectionNameModal = new SectionNameModal({model: @model})
-
       super()
-
       @listenTo(@model, 'add change:unit change:title change:expanded sync:contents', @render)
 
     onRender: () ->
+      return if @model.get('visible') == false
       super()
-
       @regions.container.empty()
-
       nodes = @model.get('contents')?.models
-
       _.each nodes, (node) =>
-        if node.isSection()
-          @regions.container.appendAs 'li', new TocSectionView
-            model: node
-        else
-          @regions.container.appendAs 'li', new TocPageView
-            model: node
-            collection: @model
+        if node.get('visible') ? true
```

Comment:
This is quite clever, but not really the easiest code to follow.  If you want to default to the `visible` property being `true`, why not just add that to the node model?